test/test_jit.py

```diff
@@ -1094,6 +1094,40 @@ def getQParamFunc(input_scale, weight_scale):
                    .check_next("_dequantize_linear").check("conv2d") \
                    .run(str(scriptModule.graph))
 
+    def test_prepare_quant(self):
+        class Observer(torch.nn.Module):
+            def __init__(self):
+                super(Observer, self).__init__()
+
+            def forward(self, x):
+                return x
+
+            @torch.jit.export
+            def calculate_qparams(self):
+                return torch.tensor([2.0]), torch.tensor([3])
+
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+                self.conv = torch.nn.Conv2d(3, 5, 3)
+
+            def forward(self, x):
+                return self.conv(x)
+
+        m = torch.jit.script(M())
+        observer = torch.jit.script(Observer())
+        torch._C._jit_pass_constant_propagation(m.graph)
```

Comment:
I'd rather explicitly specify in which graph we're running const-prop (`m._c._get_method("forward").graph`). Also, in this test we probably don't even need to run it.

Reply:
we do assume the graph we modify is after constant prop, it might work without constprop in this test case though.

Reply:
actually, this is required. otherwise we will have many observers for redundant nodes.